pop-api/integration-tests/src/nonfungibles/mod.rs

```diff
@@ -318,10 +319,26 @@ fn get_attribute_works() {
 
 #[test]
 fn next_collection_id_works() {
+	use pallet_api::{nonfungibles::*, Read as _};
 	new_test_ext().execute_with(|| {
 		let addr = instantiate(CONTRACT, INIT_VALUE, vec![]);
-		assert_eq!(next_collection_id(&addr), Ok(COLLECTION + 1));
-		assert_eq!(next_collection_id(&addr), Ok(NextCollectionId::get().unwrap_or_default() + 1))
+		assert_eq!(next_collection_id(&addr), Ok(CollectionId::default()));
+		assert_eq!(next_collection_id(&addr), Ok(NextCollectionId::get().unwrap_or_default()));
+		assert_eq!(NextCollectionId::get(), None);
+		assert_eq!(
+			NonFungibles::read(Read::NextCollectionId),
+			ReadResult::NextCollectionId(Some(0))
```

Comment:
This was the root cause and why the previous tests asserting that the next collection id was one were incorrect. The value returned is Some(0) due to the initial_value() implementation , which was being encoded to one as the return type wasnt handled properly.

To be explicit, a Some(..) value on an option encodes to 1.